lib/primo/pnxs.rb

```diff
@@ -89,16 +89,16 @@ def validators
           [{ query: :has_valid_query?,
             message: lambda { |p| "field :q must be a valid instance of Primo::Pnxs::Query " } },
           { query: :only_known_parameters?,
-            message: lambda { |p| "field :q is required " } },
+            message: lambda { |p| "only known parameters can be passed " } },
           ]
         end
 
         def has_valid_query?(params)
-          params[:q].instance_of?(Primo::Pnxs::Query)
+          params[:q].is_a?(Primo::Pnxs::Query)
         end
 
         def only_known_parameters?(params)
-          (params.keys - PARAMETER_KEYS).empty?
+          (params.keys - PARAMETER_KEYS - PARAMETER_KEYS.map(&:to_s)).empty?
```

Comment:
Could you just add the string keys to the PARAMETER_KEYS array?

Reply:
This is easier to maintain because we don't have to remember to duplicate each key in that array.

Reply:
:+1: